Web Quiz Engine (Kotlin)/task/src/engine/WebQuizApiController.kt

```diff
@@ -1,69 +1,46 @@
-// inclue repsonse et request body
-// pas de logique dans le controller
-
-
-data class AnswerQuizRequestBody(val answer: List<Int>?)
 @RestController
-class WebQuizApiController {
-
-    @Autowired
-    private lateinit var quizRepository: WebQuizRepository
-
-    sealed class QuizPostResponseBody(val success: Boolean, val feedback: String)
-    class QuizPostResponseBodyCorrect(success: Boolean = true, feedback: String = "Congratulations, you're right!")
-        : QuizPostResponseBody(success, feedback)
-    class QuizPostResponseBodyIncorrect(success: Boolean = false, feedback: String = "Wrong answer! Please, try again.")
-        : QuizPostResponseBody(success, feedback)
+class WebQuizApiController(private val quizService: QuizService) {
 
     @PostMapping("/api/quizzes/{quizId}/solve")
     fun answerQuiz(@RequestBody reqBody: AnswerQuizRequestBody, @PathVariable quizId: Int):
             ResponseEntity<QuizPostResponseBody> =
-        quizRepository.getQuizById(id = quizId)?.let { quiz ->
-            ResponseEntity(
-                if ((quiz.answerList.isNullOrEmpty() && reqBody.answer.isNullOrEmpty()) ||
-                    (quiz.answerList?.containsAll(reqBody.answer.orEmpty()) == true &&
-                            reqBody.answer?.containsAll(quiz.answerList) == true))
-                    QuizPostResponseBodyCorrect()
-                else
-                    QuizPostResponseBodyIncorrect(),
-                HttpStatus.OK,
-            )
-        } ?: throw QuizNotFoundException(quizId)
+        if (quizService.isQuizSolution(
+                quizId = quizId,
+                answerList = reqBody.answerList)) {
+            ResponseEntity(QuizPostResponseBodyCorrect(), HttpStatus.OK)
+        } else {
+            ResponseEntity(QuizPostResponseBodyIncorrect(), HttpStatus.OK)
+        }
 
     @PostMapping("/api/quizzes")
     fun addQuiz(@Valid @RequestBody quiz: Quiz): ResponseEntity<Quiz> =
         ResponseEntity(
-            quizRepository.addQuiz(quiz),
+            quizService.addQuiz(quiz = quiz),
             HttpStatus.OK,
         ).apply {
             println("Quiz added: $quiz")
         }
 
     @GetMapping("/api/quizzes/{id}")
     fun getQuizById(@PathVariable(name = "id") quizId: Int): ResponseEntity<Quiz> =
-        quizRepository.getQuizById(quizId)?.let {
-            ResponseEntity(
-                quizRepository.getQuizById(quizId, omitAnswerList = true),
-                HttpStatus.OK,
-            )
-        }?: throw QuizNotFoundException(quizId)
+        ResponseEntity(
+            quizService.findQuizById(quizId, omitAnswerList = true),
+            HttpStatus.OK,
+        )
 
     @GetMapping("/api/quizzes")
     fun getQuizList(): ResponseEntity<List<Quiz>> =
             ResponseEntity(
-                quizRepository.getAllQuizList(omitAnswerList = true),
+                quizService.findQuizList(omitAnswerList = true),
                 HttpStatus.OK,
```

Comment:
Apply same ResponseStatus annotation as metioned above

Reply:
Awesome trick. Done here too, thks for underlining them all.